### PR TITLE
ceph: add labels for CSI deployments/daemonsets/statefulsets

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: Song Gao <disxiaofei@163.com>

**Description of your changes:**
Add `rook-version` label for CSI deployments/daemonsets/statefulsets.

**Which issue is resolved by this Pull Request:**
Resolves https://github.com/rook/rook/issues/4120

**Checklist:**

- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

### CODE REVIEW

File: pkg/operator/ceph/csi/spec.go [hunk 1]
```diff
@@ -283,6 +285,7 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 		if err != nil {
 			return fmt.Errorf("failed to start cephfs plugin daemonset: %+v\n%+v", err, cephfsPlugin)
 		}
+		k8sutil.AddRookVersionLabelToDaemonSet(cephfsPlugin)
 	}
 
 	if cephfsProvisionerSTS != nil {
```
Comment (Madhu-1) [inline pkg/operator/ceph/csi/spec.go]: you need to add this to provisioner statefulset also

Comment (Yisaer) [reply pkg/operator/ceph/csi/spec.go]: > you need to add this to provisioner statefulset also

Resolved.

File: cluster/charts/rook-ceph/templates/deployment.yaml [hunk 2]
```diff
@@ -6,6 +6,7 @@ metadata:
     operator: rook
     storage-backend: ceph
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
+    rook-version: {{ .Values.image.tag }}
```
Comment (BlaineEXE) [inline cluster/charts/rook-ceph/templates/deployment.yaml]: The `rook-version` label should come from `github.com/rook/rook/pkg/version.Version` applied by the Rook operator and not from the name of the image as shown here. Remove this line.

File: pkg/operator/discover/discover.go [hunk 3]
```diff
@@ -207,6 +207,7 @@ func (d *Discover) createDiscoverDaemonSet(namespace, discoverImage, securityAcc
 	} else {
 		logger.Infof("rook-discover daemonset started")
 	}
+	k8sutil.AddRookVersionLabelToDaemonSet(ds)
```
Comment (BlaineEXE) [inline pkg/operator/discover/discover.go]: Discover daemon is not a CSI resource. This should be a change in a separate PR.
